File: tools-starter-swagger/src/main/java/cn/gjing/tools/swagger/core/SwaggerSelector.java
package cn.gjing.tools.swagger.core;

import cn.gjing.tools.swagger.Contact;
import cn.gjing.tools.swagger.SwaggerBean;
import cn.gjing.tools.swagger.SwaggerResources;
import org.springframework.context.annotation.ImportSelector;
import org.springframework.core.type.AnnotationMetadata;
import org.springframework.lang.NonNull;

/**
 * @author Gjing
 **/
public class SwaggerSelector implements ImportSelector {
    @Override
    @NonNull
    public String[] selectImports(@NonNull AnnotationMetadata annotationMetadata) {
        return new String[]{
                SwaggerBean.class.getName(),
                Contact.class.getName(),
                SwaggerConfig.class.getName(),
                SwaggerResources.class.getName(),
                ResourcesRegister.class.getName()
        };
    }
}

File: tools-starter-swagger/src/main/java/cn/gjing/tools/swagger/core/SwaggerSelector.scala
package cn.gjing.tools.swagger.core

import cn.gjing.tools.swagger.{Contact, SwaggerBean, SwaggerResources}
import org.springframework.context.annotation.ImportSelector
import org.springframework.core.`type`.AnnotationMetadata

/**
 * @author Gjing
 **/
class SwaggerSelector extends ImportSelector {
  override def selectImports(annotationMetadata: AnnotationMetadata): Array[String] = {
    Array[String](classOf[SwaggerBean].getName, classOf[SwaggerResources].getName, classOf[Contact].getName,
      classOf[SwaggerConfig].getName, classOf[ResourcesRegister].getName)
  }
}
